core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/ElementDestinationSelector.java

```diff
@@ -133,6 +135,42 @@ public boolean isNeverEnding() {
         return randomSelection || entitySelector.isNeverEnding() || valueSelector.isNeverEnding();
     }
 
+    public ListVariableDescriptor<Solution_> getVariableDescriptor() {
+        return listVariableDescriptor;
+    }
+
+    public EntityDescriptor<Solution_> getEntityDescriptor() {
+        return entitySelector.getEntityDescriptor();
+    }
+
+    public Iterator<Object> endingIterator() {
+        return Stream.concat(
+                StreamSupport.stream(Spliterators.spliterator(entitySelector.endingIterator(),
+                        entitySelector.getSize(), 0), false),
+                StreamSupport.stream(Spliterators.spliterator(valueSelector.endingIterator(null),
+                        valueSelector.getSize(), 0), false))
+                .iterator();
```

Comment:
I'm thinking that this shows up on the hot path.
If I'm correct, then I definitely wouldn't use so much stream logic for this. 
It will kill performance.

But let's worry about that later.